### PR TITLE
Feature collapsing toolbar to save space

### PR DESCRIPTION
**Pull request by osodebailar**
As a new feature the Top toolbar can now be scrolled to save space , especially a good thing for smaller displays.
To enable scrolling the <Scroll> Element is replaced by <NestedScrollView> in all fragments. 

Open for discussion or merge if you like

toolbar_expanded
toolbar_scrolled

### CODE REVIEW

File: app/src/main/java/info/nightscout/androidaps/MainActivity.kt [hunk 1]
```diff
@@ -247,6 +248,10 @@ class MainActivity : NoSplashAppCompatActivity() {
                 tab.text = (binding.mainPager.adapter as TabPageAdapter).getPluginAt(position).name
             }.attach()
         }
+
+        //appbar
```
Comment (Andries-Smit) [inline app/src/main/java/info/nightscout/androidaps/MainActivity.kt:252]: Please make comment, a single line

File: app/src/main/java/info/nightscout/androidaps/utils/ScrollingOffsetListener.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+class ScrollingOffsetListener(
+    private val scrollView: ViewPager2
+): AppBarLayout.OnOffsetChangedListener {
+
+    private var originalHeight = 0
+    private var firstOffset = true
+
+    override fun onOffsetChanged(layout: AppBarLayout?, offset: Int) {
+        if(firstOffset) {
```
Comment (Andries-Smit) [inline app/src/main/java/info/nightscout/androidaps/utils/ScrollingOffsetListener.kt:14]: Please apply code-formatting on the new/changed code (CTRL+ALT+L on Windows or Command+Option+L on Mac.)
If you want to use only a selection, you can use  CTRL+SHIFT+ALT+L on Windows

File: app/src/main/res/layout/activity_logsetting.xml [hunk 3]
```diff
@@ -82,4 +82,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/activity_logsetting.xml:85]: Please end a file with an empty newline

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -1,60 +1,73 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.drawerlayout.widget.DrawerLayout  xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:orientation="vertical"
+    android:id="@+id/main_drawer_layout"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:fitsSystemWindows="false"
+    >
```
Comment (Andries-Smit) [inline app/src/main/res/layout/activity_main.xml:8]: Code formating, will move the > one line up

File: automation/src/main/res/layout/automation_dialog_choose_operation.xml [hunk 5]
```diff
@@ -48,4 +48,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline automation/src/main/res/layout/automation_dialog_choose_operation.xml:51]: End with empty newline

File: automation/src/main/res/layout/automation_dialog_choose_trigger.xml [hunk 6]
```diff
@@ -48,4 +48,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline automation/src/main/res/layout/automation_dialog_choose_trigger.xml:51]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: automation/src/main/res/layout/automation_dialog_edit_trigger.xml [hunk 7]
```diff
@@ -49,4 +49,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline automation/src/main/res/layout/automation_dialog_edit_trigger.xml:52]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: insight/src/main/res/layout/activity_insight_pairing_information.xml [hunk 8]
```diff
@@ -156,4 +156,4 @@
             android:onClick="deletePairing"
             android:text="@string/delete_pairing" />
     </LinearLayout>
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline insight/src/main/res/layout/activity_insight_pairing_information.xml:159]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: insight/src/main/res/layout/local_insight_fragment.xml [hunk 9]
```diff
@@ -48,4 +48,4 @@
             android:visibility="gone" />
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline insight/src/main/res/layout/local_insight_fragment.xml:51]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: rileylink/src/main/res/layout/rileylink_status_general.xml [hunk 10]
```diff
@@ -423,6 +423,6 @@
 
         </LinearLayout>
 
-    </ScrollView>
+    </androidx.core.widget.NestedScrollView>
 
 </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (Andries-Smit) [inline rileylink/src/main/res/layout/rileylink_status_general.xml:428]:
```suggestion
</androidx.constraintlayout.widget.ConstraintLayout>

```

File: openhumans/src/main/res/layout/fragment_open_humans_new.xml [hunk 11]
```diff
@@ -71,4 +71,4 @@
             android:layout_marginTop="16dp"
             android:text="@string/logout" />
     </LinearLayout>
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline openhumans/src/main/res/layout/fragment_open_humans_new.xml:74]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: automation/src/main/res/layout/automation_dialog_event.xml [hunk 12]
```diff
@@ -195,4 +195,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline automation/src/main/res/layout/automation_dialog_event.xml:198]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: automation/src/main/res/layout/automation_dialog_action.xml [hunk 13]
```diff
@@ -56,4 +56,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline automation/src/main/res/layout/automation_dialog_action.xml:59]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/dialog_wizardinfo.xml [hunk 14]
```diff
@@ -496,4 +496,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/dialog_wizardinfo.xml:499]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/dialog_treatment.xml [hunk 15]
```diff
@@ -133,4 +133,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/dialog_treatment.xml:136]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/dialog_tempbasal.xml [hunk 16]
```diff
@@ -159,4 +159,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/dialog_tempbasal.xml:162]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/dialog_loop.xml [hunk 17]
```diff
@@ -404,4 +404,4 @@
             android:textAlignment="textEnd" />
 
     </LinearLayout>
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/dialog_loop.xml:407]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/dialog_extendedbolus.xml [hunk 18]
```diff
@@ -122,4 +122,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/dialog_extendedbolus.xml:125]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/dialog_calibration.xml [hunk 19]
```diff
@@ -90,4 +90,4 @@
 
     </LinearLayout>
 
-</ScrollView>
\ No newline at end of file
+</androidx.core.widget.NestedScrollView>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/dialog_calibration.xml:93]:
```suggestion
</androidx.core.widget.NestedScrollView>

```

File: app/src/main/res/layout/activity_main.xml [hunk 20]
```diff
@@ -1,60 +1,73 @@
...
+    android:id="@+id/main_navigation_view"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent"
+    app:itemBackground="?attr/selectableItemBackground"
+    android:fitsSystemWindows="true"
+    android:layout_gravity="start"
+    app:itemIconTint="?android:textColorPrimary"
+    app:itemTextColor="?android:textColorPrimary"/>
+
+</androidx.drawerlayout.widget.DrawerLayout>
```
Comment (Andries-Smit) [inline app/src/main/res/layout/activity_main.xml:73]:
```suggestion
</androidx.drawerlayout.widget.DrawerLayout>

```

**Review by Andries-Smit — CHANGES_REQUESTED** [review]
Some minor code formatting requests to change

**Comment by osodebailar** [comment]
@Andries-Smit how can this happen ?
I replace tags with file Menü : edit->find->replace .

File: app/src/main/java/info/nightscout/androidaps/MainActivity.kt (hunk 1 above)
Comment (osodebailar) [reply app/src/main/java/info/nightscout/androidaps/MainActivity.kt:251]: Reformat all files and pushed it

**Comment by osodebailar** [comment]
Reformat the code and add scrolling flag.
@MilosKozak if you found it useful we can. Merge .

**Comment by Andries-Smit** [comment]
@osodebailar 
- padding on the bottom seems to be too much
image
- I don't see the `app:layout_scrollFlags="scroll|enterAlways"` behaviour. It always scrolls out and only scrolls in at the top
- Is the scrolling offset listen needed? https://github.com/nightscout/AndroidAPS/pull/1632/files#diff-b72dad13ff175f518b8a844eb878860ba7e4eb41f2a89a7740295c87e07052deR253 I don't see it the samples of https://medium.com/@tonia.tkachuk/appbarlayout-scroll-behavior-with-layout-scrollflags-2eec41b4366b
- What is the correct behaviour when swiping? Should the top bar show when swiping, so the user knows where it goes?

**Comment by Andries-Smit** [comment]
@osodebailar From a functional perspective. 
Yes, this feature saves space, but is this required? What will be the benefit? And what is the downside? is this only for one screen, or will it affect other screens as well?  Will the navigation toolbar be "missed" at essential times?

**Comment by osodebailar** [comment]
> @osodebailar
> 
> * padding on the bottom seems to be too much
>   image
> * I don't see the `app:layout_scrollFlags="scroll|enterAlways"` behaviour. It always scrolls out and only scrolls in at the top
> * Is the scrolling offset listen needed? https://github.com/nightscout/AndroidAPS/pull/1632/files#diff-b72dad13ff175f518b8a844eb878860ba7e4eb41f2a89a7740295c87e07052deR253 I don't see it the samples of https://medium.com/@tonia.tkachuk/appbarlayout-scroll-behavior-with-layout-scrollflags-2eec41b4366b
> * What is the correct behaviour when swiping? Should the top bar show when swiping, so the user knows where it goes?

Yes I also don’t see the behaviour, but it’s not necessary I think. The user knows it after he does it once .
It’s kind of intuitive .

and without the listener the height of the view pager is wrong . In stack overflow i found the solution .

**Comment by osodebailar** [comment]
> @osodebailar From a functional perspective.
> Yes, this feature saves space, but is this required? What will be the benefit? And what is the downside? is this only for one screen, or will it affect other screens as well? Will the navigation toolbar be "missed" at essential times?

The handling is intuitive and the user will nothing essential missing . 
And required? In my experience , yes .

**Comment by MilosKozak** [comment]
I don't think this is necessary. I prefer main controls visible all the time

**Pull request closed without merging**